libkbfs/key_manager_test.go

```diff
@@ -2019,7 +2019,7 @@ func testKeyManagerRekeyAddDeviceWithPromptViaFolderAccess(t *testing.T, ver Met
 		defer ops.mdWriterLock.Unlock(lState)
 
 		// Now cause a paper prompt unlock via a folder access
-		errCh := make(chan error)
+		errCh := make(chan error, 1)
```

Comment:
Ah, it's actually not this which I think may fix the hang, but the conversion to errGroup. This is still wanted, since we don't want to leak this goroutine in case the main goroutine stops before reading from `errCh`.

I don't actually remember exactly what I was thinking on Friday, but based on the most recent run in https://keybase.atlassian.net/browse/KBFS-1641 , I had an idea that one of the MD processor goroutines were hanging, causing the test to hang. I figured the conversion to errgroup couldn't hurt, and there might have been a subtle bug with the manual err handling (although that seems unlikely come to think of it).

However, I did find another problem when looking at this today, namely why this test was hanging the whole executable even with the per-test timeout. Shutdown was using `context.TODO()`, and the state checker was going back into `MDOps` and getting wedged. Now I plumbed through the ctx through to the state checker, so hopefully in the future this will just fail the single test.